gdal/apps/ogr2ogr_lib.cpp

```diff
@@ -1455,6 +1468,19 @@ OGRFeature* GDALVectorTranslateWrappedLayer::TranslateFeature(
                     m_poFDefn->GetGeomFieldDefn(i)->GetSpatialRef() );
         }
     }
+    if (m_poOptions->bEmptyStrAsNull) {
```

Comment:
Have you tested this :-) ? This code path is only used in a very specific case: when writing to the GMLAS driver.  Furthermore, it seems you've added a check that actually errors out earlier in that situation ;-)

You want to do this logic after the poDstFeature->SetFrom() call of line 4683

Reply:
Thanks @rouault, seems I really needed that guidance.